frontend/pages/admin/IntegrationsPage/cards/MdmSettings/ScepPage/ScepPage.tsx

```diff
@@ -0,0 +1,379 @@
+                      to the SCEP admin page.
+                    </>
+                  }
+                  value={formData.password || ""}
+                  type="password"
+                  onChange={onInputChange}
+                  parseTarget
+                  placeholder="••••••••"
+                  blockAutoComplete
+                />
+                <Button
+                  type="submit"
+                  variant="brand"
+                  className="button-wrap"
+                  isLoading={isSaving}
+                  disabled={saveButtonDisabled}
+                >
+                  Save
+                </Button>
+              </form>
+            </Card>
+          </li>
+          <li>
+            <span>
+              Now head over to{" "}
+              <CustomLink
+                url={PATHS.CONTROLS_CUSTOM_SETTINGS}
+                text="Controls > OS Settings > Custom settings"
+              />{" "}
+              to configure how SCEP certificates are delivered to your hosts.{" "}
+              <CustomLink
+                url="https://fleetdm.com/learn-more-about/setup-ndes"
+                text="Learn more"
+                newTab
+              />
+            </span>
+          </li>
+        </ol>
+      </div>
+    </>
+  );
+};
+
+interface IScepPageProps {
+  router: InjectedRouter;
+}
+
+interface INdesFormData {
+  scepUrl: string;
+  adminUrl: string;
+  username: string;
+  password: string;
+}
+
+interface INdesFormErrors {
+  scepUrl?: string | null;
+  adminUrl?: string | null;
+}
+
+export interface IFormField {
+  name: string;
+  value: string;
+}
+
+const ScepPage = ({ router }: IScepPageProps) => {
+  const { isPremiumTier, config, setConfig } = useContext(AppContext);
+
+  const { renderFlash } = useContext(NotificationContext);
+
+  const [formData, setFormData] = useState<INdesFormData>({
+    scepUrl: config?.integrations.ndes_scep_proxy?.url || "",
+    adminUrl: config?.integrations.ndes_scep_proxy?.admin_url || "",
+    username: config?.integrations.ndes_scep_proxy?.username || "",
+    password: config?.integrations.ndes_scep_proxy?.password || "",
+  });
+
+  const [formErrors, setFormErrors] = useState<INdesFormErrors>({});
+  const [saveButtonDisabled, setSaveButtonDisabled] = useState(true);
+  const [isUpdatingNdesScepProxy, setIsUpdatingNdesScepProxy] = useState(false);
+
+  const {
+    data: appConfig,
+    isLoading: isLoadingAppConfig,
+    refetch: refetchConfig,
+    isError: isErrorAppConfig,
+  } = useQuery<IConfig, Error, IConfig>(["config"], () => configAPI.loadAll(), {
+    select: (data: IConfig) => data,
+    onSuccess: (data) => {
+      setConfig(data);
+    },
+  });
+
+  useEffect(() => {
+    const areAllFieldsEmpty = Object.values(formData).every(
+      (val) => val === ""
+    );
+    const areAllFieldsComplete = Object.values(formData).every(
+      (val) => val !== ""
+    );
+
+    setSaveButtonDisabled(!areAllFieldsEmpty && !areAllFieldsComplete);
```

Comment:
Instead of the `saveButtonDisabled` state and this `useEffect`, we could also set directly a
variable that depends on `formData` and disables the save button:

```
  const disableSave =
    // all fields aren't empty
    !Object.values(formData).every((val) => val === "") &&
    // all fields aren't complete
    !Object.values(formData).every((val) => val !== "");
```

and then update the `disabled` prop of the save button to be `disabled={disableSave}`

which saves us a `useState` and `useEffect`